src/ostree/ot-main.c

```diff
@@ -288,6 +288,13 @@ ostree_run (int argc, char **argv, OstreeCommand *commands, GError **res_error)
       command++;
     }
 
+  int parent_mountns = open ("/proc/self/ns/mnt", O_RDONLY | O_NOCTTY | O_CLOEXEC);
```

Comment:
The problem with this is that it will also need to be duplicated in bootc (if we were to use this as a shared library, which maybe we don't need to).

The more complex way to do this would be to have the library fork/exec itself, but that may not be worth it.

We have a flag for "mount namespace is unshared" and perhaps we should assert if it's set?